substra/sdk/client.py

```diff
@@ -78,19 +217,74 @@ class Client:
 
     def __init__(
         self,
+        *,
+        configuration_name: Optional[str] = None,
+        configuration_file: Optional[pathlib.Path] = None,
         url: Optional[str] = None,
         token: Optional[str] = None,
-        retry_timeout: int = DEFAULT_RETRY_TIMEOUT,
-        insecure: bool = False,
-        backend_type: schemas.BackendType = schemas.BackendType.REMOTE,
+        username: Optional[str] = None,
+        password: Optional[str] = None,
+        retry_timeout: Optional[int] = None,
+        insecure: Optional[bool] = None,
+        backend_type: Optional[schemas.BackendType] = None,
     ):
-        self._retry_timeout = retry_timeout
-        self._token = token
+        if configuration_file and not configuration_name:
+            logger.info("Configuration file ignored because no `configuration_name` was given.")
```

Comment:
Is there a use case where it could be interesting for a user to pass a file without a config name ? If not, might be better to raise an user friendly error here ?

Reply:
I know that in SubstraFL we have the really explicit reflex on error, so it would be `ConfigurationNameNotFoundError`, smthg like that